[고준영]ProductListView, ProductView

jay95ko opened the pull request:
상품의 리스트를 받아올 수 있는 endpoint 구현
상품 하나의 정보만 받아오는 endpoint 구현(추후에 디테일 정보를 가져달 때 같이 활용)

>wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약
(여기에 수정 사항에 대한 간략한 한줄 요약/제목 작성해 주세요.)
- ProductListView 구현
- ProductView 구현
		 
## 수정 사항들 자세한 내용
(여기에 수정 사항에 대한 자세한 내용을 작성해 주세요.)
- 상품 리스트를 받아올 수 있는(조건부여 가능) endpoint 구현(queryparameter 사용)
- 상품하나의 정보만 가져오는 endpoint구현(리스트를 받아오는 endpoint와 같은 내용을 갖지만 pathparameter 사용)

## 기타 질문 및 특이 사항
(궁금한 사항들, 하면서 느낀 점들 공유해주실 것이 있으면 자유롭게 작성해 주세요.)

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

jay95ko committed:
ProductListView, ProductView
상품의 리스트를 받아올 수 있는 endpoint 구현
상품 하나의 정보만 받아오는 endpoint 구현(추후에 디테일 정보를 가져달 때 같이 활용)

jay95ko committed:
Add: urls.py, migrations

jay95ko committed:
Add: 어울리는 안주별 검색

lordmyshepherd commented on products/views.py:
```diff
@@ -1,3 +1,99 @@
+            } for product in products]
+    return result
+
+class ProductView(View):
```
Product 객체 1개를 반환하는 엔드포인트인데 결과가 리스트로 되어 있네요!
하나만 반환될 수 있도록 수정해주세요~

jay95ko replied:
product 객체 1개만 반환하는 엔드포인트와, products라는 여러가지 객체르 반환하는 엔드포인트를 구현해주는 함수를 같이 사용하고 있어 리스트로 결과가 나오고 있습니다.
혹시 함수를 없에고 각각의 결과를 응답하도록 해야할까요?

lordmyshepherd commented on products/views.py:
```diff
@@ -1,3 +1,99 @@
+
+from .models import Category, Product
+
+def MakingList(queryset):
```
함수는 동사 시작해야하며 snake_case를 사용하셔야 합니다! 파이썬 컨벤션에 맞춰 수정해주세요~
`make_list`, `def serialize` …

lordmyshepherd commented on products/views.py:
```diff
@@ -1,3 +1,99 @@
+            SIDEDISH   = request.GET.get("side-dish")
+            products   = Product.objects.all().order_by(ORDER_BY)
+
+            if CATEGORIES:
```
For문 사용하지 않고 구현할 수 있는 방법은 없나요?

lordmyshepherd commented on products/views.py:
```diff
@@ -1,3 +1,99 @@
+            if RANDOM:
+                products_queryset  = []
+                random_number_list = []
+                for num in range(0,LIMIT):
```
어떤 결과를 반환해주는 건가요?
여기도 2중 for문으로 구현해야만 하나요? 줄일 수 있는 방법은 없을까요?

lordmyshepherd commented on products/views.py:
```diff
@@ -1,3 +1,99 @@
+
+            if DEGREES:
+                query = Q()
+                for DEGREE in DEGREES.split(","):
```
마찬가지입니다~

jay95ko committed:
로직 변경
for 문 -> list comprehention
2중 for -> for을 사용하지 않는 방법으로

jay95ko commented:
말씀주신 내용 전부 수정하였습니다!

jay95ko committed:
Modify: LIMIT default value

jay95ko committed:
commit for change branch

lordmyshepherd commented on products/views.py:
```diff
@@ -1,3 +1,88 @@
+            products   = Product.objects.all().order_by(ORDER_BY)
+
+            if CATEGORIES:
+                query = Q()
```
```suggestion
                query = Q(category__name__in=["약주", "청주", "탁주"])
```

lordmyshepherd commented on products/views.py:
```diff
@@ -1,3 +1,88 @@
+class ProductListView(View):
+    def get(self, request):
+        try:
+            ORDER_BY   = request.GET.get("order-by", "id")
```
?ordering=?

lordmyshepherd commented on products/views.py:
```diff
@@ -1,3 +1,88 @@
+                products_queryset = products.order_by("?")[:LIMIT]
+                result = make_list(products_queryset)
+
+                return JsonResponse({"Result": result}, status=200)
```
```suggestion
```

jay95ko commented:
수정 완료 하였습니다!

jay95ko committed:
로직변경
for문 최소화
result 하나일 때와 여러가지일때 분리

jay95ko committed:
Modify
queryparameter 자체가 category=청주&category=약주 이렇게 받아오는것이 아니라
category=약주,청주 로 받기에 split 사용

jay95ko committed:
Category를 name값이 아닌 id값으로 검색

lordmyshepherd reviewed (approved):
LGTM 👍

jay95ko committed:
fix conflict
